### ISSUE TITLE
IDE0079 triggering false positives on various suppressed diagnostics

### ISSUE THREAD

**Issue by DanielRouxSA** [dotnet/roslyn#75887]
**Version Used**:

.NET 9 SDK 9.0.100 (though this also occurs on projects targeting .NET 8)

**Steps to Reproduce**:

Since the update to Visual Studio 17.12.0 (and thus to the .NET 9 SDK), I am now experiencing false positives for IDE0079 on numerous different analyzer suppressions across my projects.

Below I have a few selected minimal examples. Further down I will list the full list of diagnostic suppressions that I have encountered with false positives for IDE0079.

*CA1040*:
```CS
// For context, the below interface is solely used for polymorphic serialization (as per System.Text.Json requirement).

[SuppressMessage("Design", "CA1040:Avoid empty interfaces")] // IDE0079 triggers here. Removing this line causes CA1040 to correctly trigger on the interface.
[JsonDerivedType(typeof(PolymorphicTypeA), "typeA")]
[JsonDerivedType(typeof(PolymorphicTypeB), "typeB")]
[JsonDerivedType(typeof(PolymorphicTypeC), "typeC")]
[JsonDerivedType(typeof(PolymorphicTypeD), "typeD")]
public interface IPolymorphicSearchResult_V4 { }
```

*CA1815:*
```CS
#pragma warning disable CA1815 // Override equals and operator equals on value types // IDE0079 triggers here. Removing this suppression causes CA1815 to correctly trigger on the struct.

file readonly struct GuidResponse
{
    #region JSON Properties

    [JsonPropertyName("Response")] public Guid response { get; init; }

    #endregion
}

#pragma warning restore CA1815 // Override equals and operator equals on value types
```

*CA1822:*
```CS
[SuppressMessage("Performance", "CA1822:Mark members as static")] // IDE0079 triggers here. Removing this line causes CA1822 to correctly trigger on the method.
internal void ValidateNewInstruction(NewInstruction instruction)
{
    // Method implementation that does not access instance data
}
```

**Diagnostic Id**:

IDE0079 is triggered.

This is the full list of analyzer suppressions that I have (thus far) encountered that cause the IDE0079 false positives:

- CA1002
- CA1008
- CA1024
- CA1034
- CA1040
- CA1054
- CA1056
- CA1308
- CA1310
- CA1711
- CA1815
- CA1819
- CA1822
- CA1849
- CA1859
- CA5394
- VSTHRD100 (From the Visual Studio Threading Analyzers)
- VSTHRD103 (From the Visual Studio Threading Analyzers)
- VSTHRD200 (From the Visual Studio Threading Analyzers)

I have also seen a number of analyzers that do not trigger IDE0079 when suppressed, so this does not occur on **all** suppressions.

Just to be clear, there may well be cases among these analyzers where my way of handling them could be improved as opposed to merely suppressing them; this issue is specifically regarding the false positives for IDE0079 on said suppressions.

**Expected Behavior**:

IDE0079 does not trigger on suppressions that suppress otherwise active analyzer triggers.

**Actual Behavior**:

IDE0079 triggers on suppressions that suppress otherwise active analyzer triggers.

**Comment by rampaa** [comment]
I'm also encountering the same issue when using .NET 9. In addition to the aforementioned rules, I'm also getting false positives for the following rules: `CA2100`, `CA2227`, `CA5351`, and `CA1812`. For context, I've set `AnalysisLevel` to `9-all` and changed the default severity of all rules to `warning` by configuring it in the .editorconfig file like so: `dotnet_analyzer_diagnostic.severity = warning`.

**Comment by johnstrand** [comment]
Same issue with CA2254 and SYSLIB1045

**Comment by arunchndr** [comment]
There was a fix that went out for this in 17.12.3 tracked via https://developercommunity.visualstudio.com/t/Private-Unity-messages-incorrectly-marke/10779025. Closing as duplicate.

**Comment by abdulkareemnalband** [comment]
It is not fixed in 17.12.3

**Comment by DanielRouxSA** [comment]
I can confirm it is still occurring in 17.12.3.

In addition, @arunchndr I read through that Visual Studio Feedback ticket - to me that ticket does not appear related to this issue, so I presently disagree that this issue is a duplicate.

**Comment by abdulkareemnalband** [comment]
looks like this belongs  to #75858

**Comment by rampaa** [comment]
@arunchndr The issue is NOT fixed. I am using Visual Studio 17.13.0 and yet I'm still getting false positives for the following rules: CA5351, CA1056, CA1308, CA1812, CA5394. Kindly consider re-opening the issue.

**Comment by arunchndr** [comment]
@JoeRobich would you confirm if the analysislevel setting to behavior matches and close accordingly?

**Comment by menees** [comment]
I'm running VS 17.13.6, and I'm getting IDE0079 false positives for CA1822.

**Comment by migoldfinger** [comment]
I am running 17.14.0 still tons of false positives.

**Comment by idg10** [comment]
Same here with 17.14.4.

**Comment by HgAlexx** [comment]
Test project of false positive of IDE0079 with CA1000:
https://github.com/HgAlexx/IDE0079

Visual studio show IDE0079 in the ErrorList window when you open file ClassTest.cs but compile without issue.

**Comment by bill-poole** [comment]
I am getting this false positive with Visual Studio 17.14.14 on a variety of legitimately suppressed warnings - e.g., SYSLIB1015, CA2012.
